plugins/search/src/components/SidebarSearchModal/SidebarSearchModal.tsx

```diff
@@ -13,30 +13,42 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import React from 'react';
+import React, { useCallback, useState } from 'react';
 import SearchIcon from '@material-ui/icons/Search';
 import { SidebarItem } from '@backstage/core-components';
 import { IconComponent } from '@backstage/core-plugin-api';
 import { SearchModal } from '../SearchModal';
-import { useSearch } from '../SearchContext';
 
 export type SidebarSearchModalProps = {
   icon?: IconComponent;
 };
 
 export const SidebarSearchModal = (props: SidebarSearchModalProps) => {
-  const { open, toggleModal } = useSearch();
+  const [state, setState] = useState({ hidden: true, opened: false });
```

Comment:
I don't think this was currently possible before this PR. The problem is, in order to use `toggleModal` on a different place, the context needs to be rendered. This was the reason why I've removed from the context.
But of course, if this is a need, we could still keep `open`, `toggleModal` in the context and render `SearchContextProvider` in a more generic place, like `App.tsx`.

What do you think?

Reply:
I asked because I remembered [this](https://github.com/backstage/backstage/pull/8302) pull request, I believe we added it out of internal necessity, let me involve @emmaindal  here to confirm with her if it's ok to remove the context (I believe it will need to change our internal Backstage instance)

Reply:
@emmaindal  and I have talked and If we wanted to have just a global instance of modal we would have to move the context to wrap the entire application and that would affect the context of the search page, so we are fine to remove the context. The only repetitive thing is that we have to use the modal component and set a state for it whenever we want to open the modal from elsewhere...  What about having a hook that creates and initializes a state for the modal? For example:
```jsx
const HomePage = () => {
 // Can receive the initial state useSearchModal(true), but the default is false
const { open, setOpen, toggleModal } = useSearchModal();

return (
   <Page>
     ...
     <Input placeholder="Search" onClick={toggleModal} />
     <SearchModal open={open} />
     ...
   <Page>
   );
};
```

Reply:
This documentation also needs to be updated to remove Context from the examples:
https://github.com/backstage/backstage/blob/d11fb7fe043183ce6f6d5e994dd7749ea609e938/docs/features/search/getting-started.md

Reply:
I've just realized that global rendering the SearchContextProvider will not work, since it will re-introduce the problem of fetching data whenever the page is loaded.

> @emmaindal and I have talked and If we wanted to have just a global instance of modal we would have to move the context to wrap the entire application and that would affect the context of the search page, so we are fine to remove the context. The only repetitive thing is that we have to use the modal component and set a state for it whenever we want to open the modal from elsewhere... What about having a hook that creates and initializes a state for the modal? For example:
> 
> ```js
> const HomePage = () => {
>   // Can receive the initial state useSearchModal(true), but the default is false
>   const { open, setOpen, toggleModal } = useSearchModal();
> 
>   return (
>     <Page>
>       ...
>       <Input placeholder="Search" onClick={toggleModal} />
>       <SearchModal open={open} />
>       ...
>     </Page>
>   );
> };
> ```

but this solution doesn't solve the problem of opening the modal from elsewhere, since every time `useSearchModal()` is invoked you get a different instance of toggleModal. Unless we make the hook rely on another `SearchModalContext` (separated from `SearchContextProvider`) rendered globally, containing only `open`, `toggleModal`.

Reply:
I'm not sure if I explained it correctly, but what we mean is that we agree with you about removing the context, we like the idea of removing it 🙂 .
The suggested solution works in a more scoped way, as the `HomePage` for example will have its own search modal and will be able to control its state only, will open/close the rendered modal inside of its children, not the modal rendered inside the Sidebar.

Reply:
sorry I misread it, makes sense :)

Reply:
ok, I think I've addressed all the issues 😄
Since `useSearchModal` returns both `open` and `hidden`, I've wrapped them in a `state` property to be spread in `SearchModal` to implicitly mark both as "needed".
Let me know if you prefer to include them in the main returned object: `return { open, hidden, toggleModal, setOpen }`